### PR TITLE
Defaults in dpl-react configuration

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-246

#### Description

This PR is created as a port of the work being done on this [PR](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/436) in dpl-cms.
Initiatives in this project:
* Changed allowRemoveReadyReservations to be boolean property. I think we should handle booleans as a part of a json structure config. In that way it is "natively boolean" and we do not have to juggle around with tranforming "1" and "0" or "true" and "false".
* Instant loan config. The place it is used there was no checks if the various config properties were set. I tried to accomodate the situation by let the handling functions accept null parameters where they were used. Not optional though would like to handle it earlier.
* In the work with the allowRemoveReadyReservations boolean I wanted to be sure of how the config handler works with the different input. So I wrote a pair of unit tests to document it.

#### Additional comments or questions

This PR has precedence over https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/436.

### CODE REVIEW

File: src/tests/unit/config.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { configureStore, combineReducers } from "@reduxjs/toolkit";
```
Comment (kasperg) [inline src/tests/unit/config.test.tsx:1]: Good stuff adding tests.

I wonder if there is a reason why this does not use in file testing?

File: src/components/reservation/helper.ts [hunk 2]
```diff
@@ -216,7 +216,7 @@ export const consolidatedHoldings = (branchHoldings: HoldingsV3[]) => {
 export const getInstantLoanBranchHoldings = (
   branchHoldings: HoldingsV3[],
   whitelist: AgencyBranch[],
-  instantLoanStrings: string[]
```
Comment (kasperg) [inline src/components/reservation/helper.ts:219]: I am not too keen on this change. As I see it it goes against [our decision](https://github.com/danskernesdigitalebibliotek/dpl-react/blob/develop/docs/architecture/adr-007-configuration.md) to make the mouting application provide configuration in the expected format.

If no strings are provided then we should at least be able to set a `{enabled: false}`. I guess this goes against the design in danskernesdigitalebibliotek/dpl-cms#436 where modules can register their own settings. 🤔 

If we have become wiser can we do something do update the ADR accordingly?

**Review by kasperg — APPROVED** [review]
:+1: but with a few considerations for you.

File: src/tests/unit/config.test.tsx (hunk 1 above)
Comment (spaceo) [reply src/tests/unit/config.test.tsx:1]: Becase "@testing-library/react-hooks" is a dev dependency and the config.ts file is a production file

**Pull request merged**